### PR TITLE
HTCONDOR-1586: boost 1.81 update

### PR DESCRIPTION
**Pull request by timtheisen**
Recommend ignoring whitespace changes

### CODE REVIEW

**Comment by timtheisen** [comment]
Either @Todd-L-Miller or @jasoncpatton can review this. Whoever get here first.

File: manylinux2014/docker/build_scripts/build_external.sh [hunk 1]
```diff
@@ -13,30 +13,26 @@ function boost {
...
             pythons="${pythons},${majv}.${minv}"
-	    fi
-	done
-	echo "}" >> tools/build/src/user-config.jam
-	./b2 install --prefix=/usr/local --layout=system -j$NPROC variant=release link=static define=BOOST_HAS_THREADS cxxflags=-fPIC linkflags=-fPIC --with-python python=${pythons:1} # strip the leftmost comma
-	#rm -rf /usr/local/lib/cmake
-	popd
+        done
+        echo "}" >> tools/build/src/user-config.jam
+        ./b2 install --prefix=/usr/local --layout=system -j$NPROC variant=release define=BOOST_HAS_THREADS cxxflags=-fPIC linkflags=-fPIC --with-python python=${pythons:1} # strip the leftmost comma
```
Comment (jasoncpatton) [inline manylinux2014/docker/build_scripts/build_external.sh:34]: static boost removed

File: manylinux2014/build_docker_image.sh [hunk 2]
```diff
@@ -4,6 +4,7 @@ set -ex
 DOCKERFILE_DIR=$(dirname "$(readlink -f "$0")")/docker/
 TAG="htcondor_manylinux2014_x86_64:$(head -n 1 latest_tag)"
 
+docker pull quay.io/pypa/manylinux2014_x86_64
```
Comment (jasoncpatton) [inline manylinux2014/build_docker_image.sh:7]: updating image in docker cache each re-build

File: README.md [hunk 3]
```diff
@@ -1,8 +1,6 @@
 # HTCondor manylinux Python Wheel Building Routines
 
 1. Choose which version of manylinux to build for. Versions currently available:
-   * `manylinux1` - Builds inside CentOS 5-based Docker container, most compatible, limits versions of some externals (e.g. kerberos, keyutils).
-   * `manylinux2010` - Builds inside CentOS 6-based Docker container.
    * `manylinux2014` - Builds inside CentOS 7-based Docker container.
```
Comment (jasoncpatton) [inline README.md:4]: only building CentOS 7+ wheels

**Review by jasoncpatton — APPROVED** [review]
LGTM... the three big changes:
1. Only building the manylinux2014 image (i.e. wheels will be built for CentOS 7-compatible or newer)
2. The underlying manylinux image is pulled each time to update the image in the Docker cache.
3. Static library for Boost.python no longer being built or used.

**Pull request merged**